libs/common/database-ui/src/hooks/useDatabaseTally.tsx

```diff
@@ -0,0 +1,16 @@
+import type { DataManagerBase } from '@genshin-optimizer/common/database'
+import { useEffect, useState } from 'react'
+
+export function useDatabaseTally(
+  dataManager: DataManagerBase<any, any, any, any, any>
+) {
+  const [total, setTotal] = useState(dataManager.keys.length)
+  useEffect(() => {
+    setTotal(dataManager.keys.length)
+    return dataManager.followAny(
+      (_k, r) =>
+        ['new', 'remove'].includes(r) && setTotal(dataManager.keys.length)
```

Comment:
Why do we have to run the set total on cleanup?

Reply:
`followAny` return a "unfollow", so basically every time you change datamanager, it would "register"(via followAny), and when the datamanager changes (goes out of scope), it would run the "unfollow", this makes it so that this component is only registered to listen to a singular datamanager, and new datamangers are automatically unfollowed(old datamanager) and followed(new datamanager).

For more, see "return a cleanup function" part of doc https://react.dev/reference/react/useEffect#useeffect

Reply:
Oh I misread the indentation on mobile, I thought we were calling setTotal in the cleanup woops